sycl/include/CL/sycl/detail/program_impl.hpp

```diff
@@ -203,12 +204,14 @@ class program_impl {
     if (!is_host()) {
       OSModuleHandle M = OSUtil::getOSModuleHandle(AddressInThisModule);
       // If there are no build options, program can be safely cached
-      if (BuildOptions.empty()) {
-        Program = ProgramManager::getInstance().getBuiltOpenCLProgram(M, Context);
+      if (is_cacheable_with_build_options(BuildOptions)) {
```

Comment:
Minor. Suggest writing something:
// Do not support caching program with non default options so far
AllowKernelsCaching &= BuildOptions.empty();
...
if(is_cacheable()) { ...

Reply:
`is_cacheable()` relies on value of `this->BuildOptions` which is not set to `BuildOptions` passed as argument to `build_with_kernel_type` method. I could add another implementation of `is_cacheable` method which will accept build options but it will just blow the source file size without any use. Still, I could assign `this->BuilOptions` prior to calling `is_cacheable()`  but it'll require unnecessary changes in other methods like `build` and possibly `compile` and `link` where another assignments to `this->BuildOptions` take place.

Reply:
Ok, I see.